pkg/kp/rcstore/store.go

```diff
@@ -0,0 +1,32 @@
+package rcstore
+
+import (
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/rc/fields"
+)
+
+// Store represents an interface for persisting replication controllers to Consul,
+// as well as restoring replication controllers from Consul.
+type Store interface {
+	// Create creates a replication controller with the specified manifest and selectors.
+	// The node selector is used to determine what nodes the replication controller may schedule on.
+	// The pod label set is:
+	// 1) applied to every pod the replication controller schedules,
+	// 2) used (in addition to replication controller ID) to determine what pods are already scheduled.
```

Comment:
I'll tentatively say that the answer is "yes, two replication controllers with the same pod labels will see each others' pods", code and comment updated to match.